lib/UR/Object/Type/AccessorWriter.pm

```diff
@@ -402,145 +402,142 @@ sub _resolve_bridge_logic_for_indirect_property {
-                                                  $result_class_meta->resolve_composite_id_from_ordered_values(
-                                                      map { $bridge->$_ } @$bridging_identifiers
-                                                  );
-                                            },
-                                            sub { $_->id } );
-                    @results = $results_sorter->(\@results);
-                }
-                return @results;
+                my $result_class = $result_class_resolver->($bridge);
+                my $id_resolver = $composite_id_resolver_for_class{ $result_class }
+                                    ||= $result_class->__meta__->get_composite_id_resolver;
+
+                return $id_resolver->(@id);
             };
-        } elsif ($to_property_meta->id_by and $to_property_meta->data_type and not $to_property_meta->data_type->isa('UR::Value')) {
-            my $result_class = $to_property_meta->data_type;
-            my $bridging_identifiers = $to_property_meta->id_by;
-            my $id_resolver = $result_class->__meta__->get_composite_id_resolver;
+        };
 
-            $bridge_crosser = sub {
-                my $bridges = shift;
-                my @ids;
-                foreach my $bridge ( @$bridges ) {
-                    my @id = map { $bridge->$_ } @$bridging_identifiers;
-                    my $id = $id_resolver->(@id);
+        return sub {
+            my $bridges = shift;
+            my %result_class_names_and_ids;
 
-                    push @ids, $id;
-                }
+            foreach my $bridge ( @$bridges ) {
+                my $result_class = $result_class_resolver->($bridge);
+                $result_class_names_and_ids{$result_class} ||= [];
 
-                my @results = (@_ ? $result_class->get(id => \@ids, @_) : $result_class->get(\@ids) );
-                if ($bridge_meta_params{'-order'} || $bridge_meta_params{'-order_by'}) {
-                    my $results_sorter = $make_results_sorter->(
-                                            $bridges,
-                                            sub { my $bridge = $_;
-                                                  $id_resolver->(
-                                                      map { $bridge->$_ } @{ $to_property_meta->id_by }
-                                                  );
-                                            },
-                                            sub { my $result = $_;
-                                                  $_->id;
-                                            } );
-                    @results = $results_sorter->(\@results);
+                my $id = $linking_id_value_for_bridge->($bridge);
+                push @{ $result_class_names_and_ids{ $result_class } }, $id;
+            }
+
+            my @results;
+            foreach my $result_class ( keys %result_class_names_and_ids ) {
+                if (@_) {
```

Comment:
You're right.  The distinction vanished after the refactoring.

Pushed a new commit to remove the redundant code